tests/conftest.py

```diff
@@ -893,33 +903,58 @@ def is_port_in_use(host: str, port: int) -> bool:
         return s.connect_ex((host, port)) == 0
 
 
+class BaseP2PLibp2pTest:
+    """Base class for p2p libp2p tests"""
+
+    cwd: str
+    t: str
+    tmp_dir: str
+    log_files: List[str] = []
+    multiplexers: List[Multiplexer] = []
+    capture_log = True
+
+    @classmethod
+    def setup_class(cls):
+        """Set the test up"""
+        cls.cwd, cls.t = os.getcwd(), tempfile.mkdtemp()
+        cls.tmp_dir = os.path.join(cls.t, "temp_dir")
+        Path(cls.tmp_dir).mkdir(exist_ok=True)
+        os.chdir(cls.t)
+        atexit.register(cls.teardown_class)
+
+    @classmethod
+    def teardown_class(cls):
+        """Tear down the test"""
+        logger.debug(f"Cleaning up {cls.__name__}")
+        for mux in cls.multiplexers:
+            mux.disconnect()
+        cls.multiplexers.clear()
+        cls.log_files.clear()
+        os.chdir(cls.cwd)
+        remove_test_directory(cls.t)
+        logger.debug(f"Teardown of {cls.__name__} successful")
```

Comment:
base class for libp2p tests. `atexit` ensures that the teardown gets called no matter what, whereas I've seen that otherwise processes don't always get terminated (`netstat -ap | grep libp2p_node`). Probably should be in a in a different location, `p2p_libp2p/helper.py` or so